packages/executors/src/lib.ts

```diff
@@ -0,0 +1,104 @@
+import { isMatch, isString } from 'lodash-es';
+
+type Executor =
+  | any
+  | ((id: string, variables: Record<string, any>) => any | Promise<any>);
+
+type VariablesMatcher =
+  | Record<string, any>
+  | ((vars: Record<string, any>) => boolean);
+
+type RegistryEntry = {
+  executor: Executor;
+  id?: string;
+  variables?: VariablesMatcher;
+};
+
+let registry: Array<any> = [];
+
+export function clearRegistry() {
+  registry = [];
+}
+
+export function registerExecutor(executor: Executor): void;
+
+export function registerExecutor(id: string, executor: Executor): void;
+
+export function registerExecutor(
+  id: string,
+  variables: VariablesMatcher,
+  executor: Executor,
```

Comment:
I thought it would produce a more natural API:
```javascript
// register an executor that matches everything
registerExecutor(iDoEverythingOverHttp);
// register one for a specific operation;
registerExecutor(MyOperationId, myOperationImplementation);
// for a specific set of input:
registerExecutor(ContentHubQuery, {page: 1}, staticFirstPage);
```
Compared to:
```javascript
// register an executor that matches everything
registerExecutor({executor: iDoEverythingOverHttp});
// register one for a specific operation;
registerExecutor({id: MyOperationId, executor: myOperationImplementation});
// for a specific set of input:
registerExecutor({id:ContentHubQuery, args: {page: 1}, executor: staticFirstPage});
```

I'm not sure if it has to be possible to infer types here, since the`Operation*` type helpers are available to infer return types and variables:
```typescript
type first = AnyOperationId;
type second = Partial<OperationVariables<first>>;
type third = () => Promise<OperationResult<first>>;
```

But I also don't have a very strong opinion on this. How strong are yours? Should we pull in more voices?